### PR TITLE
Proposal validation tests (receiver side) for ValSem100-103

### PR DESCRIPTION
**Pull request by kkohbrok**
As suggested by the title, this PR adds tests for ValSem100-ValSem103.

It also adds some test/test-utils only functions to facilitate the tests.

### CODE REVIEW

File: openmls/src/group/tests/test_proposal_validation.rs [hunk 1]
```diff
@@ -204,7 +423,84 @@ fn test_valsem101(ciphersuite: Ciphersuite, backend: &impl OpenMlsCryptoProvider
         }
     }
 
-    // TODO #525: Add test for incoming proposals.
+    // Before we can test reception of (invalid) proposals, we set up a new
```
Comment (franziskuskiefer) [inline openmls/src/group/tests/test_proposal_validation.rs:426]: Can you describe first on a high level what you're testing here? Same for the other tests.

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
Looks good, just a little more high level description for the tests would be nice.

File: openmls/src/group/tests/test_proposal_validation.rs (hunk 1 above)
Comment (kkohbrok) [reply openmls/src/group/tests/test_proposal_validation.rs]: I added a remark to each test, indicating that we're now testing the same validation logic on the receiver side. Is that what you meant?

**Review by franziskuskiefer — APPROVED** [review]
Thanks, yeah it tells the reader what's going on.

**Pull request merged**